src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -308,6 +320,42 @@ void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
             return default;
         }
 
+        private ImmutableArray<TaggedText> GetReturnsDocumentationContent(
+            IEnumerable<ISymbol> symbols,
+            IDictionary<SymbolDescriptionGroups, ImmutableArray<TaggedText>> sections,
+            SemanticModel semanticModel,
+            SyntaxToken token,
+            IDocumentationCommentFormattingService formatter,
+            ISyntaxFactsService syntaxFactsService,
+            CancellationToken cancellationToken)
+        {
+            if (sections.TryGetValue(SymbolDescriptionGroups.ReturnsDocumentation, out var parts))
+            {
+                var documentationBuilder = new List<TaggedText>();
+                documentationBuilder.AddRange(parts);
+                return documentationBuilder.ToImmutableArray();
+            }
+            else if (symbols.Any())
+            {
+                var symbol = symbols.First().OriginalDefinition;
+
+                // if generating quick info for an attribute, bind to the class instead of the constructor
+                if (syntaxFactsService.IsAttributeName(token.Parent) &&
+                    symbol.ContainingType?.IsAttribute() == true)
+                {
+                    symbol = symbol.ContainingType;
+                }
```

Comment:
Is this applicable here? (This looks copy/pasted.)

Reply:
I'll refactor this to avoid duplication.